docstr_coverage/cli.py

```diff
@@ -328,7 +337,11 @@ def execute(paths, **kwargs):
     show_progress = not kwargs["percentage_only"]
     results = analyze(all_paths, ignore_config=ignore_config, show_progress=show_progress)
 
-    LegacyPrinter(verbosity=kwargs["verbose"], ignore_config=ignore_config).print(results)
+    if kwargs["output"] == "markdown":
+        printer = MarkdownPrinter(results, verbosity=kwargs["verbose"], ignore_config=ignore_config)
+    else:
+        printer = LegacyPrinter(results, verbosity=kwargs["verbose"], ignore_config=ignore_config)
+    printer.print()
```

Comment:
What was the reasoning for moving `results` to being provided on Printer initialization, rather than on the call to `LegacyPrinter.print`/`MarkdownPrinter.print`? Was this a stylistic choice, or was it necessary for your PR?

Reply:
`results` moved to initialized because of new mechanism: collecting data to print (depending on verbose value) in base class for than using in child classes.